scipy/stats/_continuous_distns.py

```diff
@@ -7240,6 +7438,78 @@ def _stats(self, a, b, moments='mv'):
         g2 = mu4 / mu2**2 - 3
         return mu, mu2, g1, g2
 
+    def _rvs(self, a, b):
+        # if a and b are scalar, use _rvs_scalar, otherwise need to create
+        # output by iterating over parameters
+        if np.isscalar(a) and np.isscalar(b):
+            out = self._rvs_scalar(a, b, self._size)
+        elif a.size == 1 and b.size == 1:
+            out = self._rvs_scalar(a.item(), b.item(), self._size)
+        else:
+            # When this method is called, self._size will be a (possibly empty)
```

Comment:
one could remove the comments here to keep it shorter and link to _rvs of geninvgauss (Warren wrote that code when I implemented geninvgauus)

Reply:
I'm actually wondering whether long-term the `_rvs_scalar` framework could be moved into `rv_continuous` or `rv_generic`, (perhaps also supporting `_cdf_scalar`, `_sf_scalar`, ...) but that is a further-out project.

Reply:
I agree, I think that would be a useful feature since once the implementation relies on the rejection method or ratio of uniforms method, vectorization w.r.t. the shape parameters needs extra work.